framework/include/cppmicroservices/detail/BundleAbstractTracked.tpp

```diff
@@ -110,18 +112,21 @@ void BundleAbstractTracked<S,TTT,R>::Close()
 template<class S, class TTT, class R>
 void BundleAbstractTracked<S,TTT,R>::Track(S item, R related)
 {
-  std::shared_ptr<TrackedParamType> object;
+  bool isInMap = false;
```

Comment:
Just wondering if this is threadsafe. It's done outside the lock for the object.

Reply:
The flag `isInMap` is declared outside of the synchronized block since it is required later to determine which customizer to call. Since the flag is local to the function call and the actual map-checking is synchronized, it should be okay.